### ISSUE TITLE
WithNonVoidReturnType matches methods with void return type

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#966]
Specifically, this fails:

```c#

public interface IFoo
{
    void Baz();
}

[Scenario]
public static void WithNonVoidReturnTypeAndVoidMethod(
    IFoo fake,
    bool methodWasCalled)
{
    "Given a fake with a void method"
        .x(() => fake = A.Fake<IFoo>());

    "When the fake is configured to set a flag for any non-void return type"
        .x(() => A.CallTo(fake).WithNonVoidReturnType().Invokes(() => methodWasCalled = true));

    "And the void method is called"
        .x(() => fake.Baz());

    "Then the flag is not set"
        .x(() => methodWasCalled.Should().BeFalse());
}
```

Found while working on external Self-Initializing Fakes project.

**Comment by blairconrad** [comment]
`WithNonVoidReturnType` was introduced in 2.1.0, so this is a candidate for backport to the `support-2` branch.

**Comment by blairconrad** [comment]
Fixed in #971.

**Comment by blairconrad** [comment]
This issue has been releaseed in FakeItEasy 2.3.3:
https://github.com/FakeItEasy/FakeItEasy/releases/tag/2.3.3
